### PR TITLE
Performance test for `OmegaConfigLoader`

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description

Partial solution for #4125 

## Development notes

Added performance test for `OmegaConfigLoader` in `benchmarks/benchmark_ocl.py` for basic operations for loading catalog, parameters, globals
As well as another test class for operations with global and custom resolver.

Test locally with `asv run`

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: benchmarks/benchmark_ocl.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        with open(file_path, "w") as f:
+            yaml.dump(data, f)
+
+
+class TimeOmegaConfigLoader:
+
+    def setup(self):
+        # Setup temporary configuration directory with sample config files
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.conf_source = Path(self.temp_dir.name)
```
Comment (noklam) [inline benchmarks/benchmark_ocl.py:71]: It seems that `asv` already create some temp dir when I run test, do we need to create a separate one ourself?

File: benchmarks/benchmark_ocl.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        "filepath": "data5_local.csv",
+        "type": "pandas.CSVDataset",
+    },
+}
+base_params = {
+    "param_1": "value_1",
+    "param_2": "value_2",
+    "param_3": "value_3",
+    "param_4": "value_4",
+}
```
Comment (noklam) [inline benchmarks/benchmark_ocl.py:40]: Can we have some extreme test to test OmegaConfigLoader is still performing ok when dealing with a large amount of config? i.e. 10000+ configuration. I have seen big projects that has 20+ pipelines and take half a min just to load config (not with OmegaConfigLoader).

File: benchmarks/benchmark_ocl.py (hunk 1 above)
Comment (ankatiyar) [reply benchmarks/benchmark_ocl.py:64]: This is to create a temporary conf directory structure for the config for each test

File: benchmarks/benchmark_ocl.py (hunk 2 above)
Comment (ankatiyar) [reply benchmarks/benchmark_ocl.py]: Updated to larger configs

**Review by merelcht — COMMENTED** [review]
This looks great! And the main difference between  `TimeOmegaConfigLoader` and `TimeOmegaConfigLoaderAdvanced` is interpolation?

**Comment by ankatiyar** [comment]
> This looks great! And the main difference between `TimeOmegaConfigLoader` and `TimeOmegaConfigLoaderAdvanced` is interpolation?

@merelcht Yeah, I had to separate it out into a new class because the `setup()` function had to be different - `asv` runs the `setup()` and `teardown()` before each `time_` test.

**Review by noklam — APPROVED** [review]

**Review by ElenaKhaustova — APPROVED** [review]
LGTM!

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Pull request merged**